src/Kernel/IQSharpEngine.cs

```diff
@@ -342,7 +342,31 @@ public override async Task<ExecutionResult> ExecuteMundane(string input, IChanne
                 }
                 catch (CompilationErrorsException c)
                 {
-                    foreach (var m in c.Errors) channel.Stderr(m);
+                    // Check if the user likely tried to execute a magic
+                    // command and try to give a more helpful message in that
+                    // case.
+                    if (input.TrimStart().StartsWith("%") && input.Split("\n").Length == 1)
+                    {
+                        var attemptedMagic = input.Split(" ", 2)[0];
+                        var msg = $"No such magic command {attemptedMagic}.";
+                        if (MagicResolver is MagicSymbolResolver iqsResolver)
+                        {
+                            var similarMagic = iqsResolver
```

Comment:
what happens if we don't find 3 similar magics, will the message be all weird?

Reply:
From testing, it looks like the worst case is that the suggestions are a bit off-topic. The actionable message about running `%lsmagic` or checking docs.microsoft.com should help in that case, I think?